sol/src/Logger.sol

```diff
@@ -0,0 +1,1552 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.22 <0.9.0;
+
+library Logger {
+    address constant LOGGER_FFI_ADDRESS = address(0xFFCCCC);
+
+    function _sendLogPayloadImplementation(bytes memory payload) internal view {
+        address consoleAddress = LOGGER_FFI_ADDRESS;
+        /// @solidity memory-safe-assembly
+        assembly {
```

Comment:
Since it's memory safe, you can define this as `assembly ("memory-safe") {`.

Reply:
good point, though I just stole this shamelessly from `Console.log` and I want to maybe try to make them cross-compataible...

Reply:
In fact, this file is just literally now `console.sol` from forge. I decided to make them fully compatible.